webhooks/pod_webhook.go

```diff
@@ -118,22 +99,43 @@ func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admissio
 	}
 
 	// merge any provided flagd specs
+	flagSourceConfigurationSpec, code, err := m.createFSConfigSpec(ctx, req, annotations, pod)
+	if err != nil {
+		return admission.Errored(code, err)
+	}
+
+	marshaledPod, err := m.injectSidecar(ctx, pod, flagSourceConfigurationSpec)
+	if err != nil {
+		m.Log.Error(err, "unable to inject flagd sidecar")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+}
+
+func (m *PodMutator) createFSConfigSpec(ctx context.Context, req admission.Request, annotations map[string]string, pod *corev1.Pod) (*v1alpha1.FlagSourceConfigurationSpec, int32, error) {
+	// Check configuration
+	fscNames := []string{}
+	val, ok := annotations[fmt.Sprintf("%s/%s", OpenFeatureAnnotationPrefix, FlagSourceConfigurationAnnotation)]
+	if ok {
+		fscNames = parseList(val)
+	}
```

Comment:
We're doing this check in 2 places now. Does it make more sense to keep it early?

Reply:
Agreed!